In [73]:
#pip install python-docx

In [74]:
import functions
import docx
import requests 

In [86]:
from dotenv import load_dotenv
import os

# Load environment variables from a .env file
load_dotenv()

# Retrieve the API key from the environment variables
api_key = os.getenv("HUGGINGFACE_API_KEY")

In [75]:
chat_completion = functions.chat_completion

In [76]:
chat_completion ("hello")

"This instruction appears to be a straightforward greeting in English. No further action is required beyond acknowledging the greeting, as it's a typical form of communication in many contexts."

In [77]:
# Specify the file path (make sure it's correct)
file_path = '..\Incident_Reports\incident_report_1.txt'

# Open the file in read mode and read its content into a string
with open(file_path, 'r') as file:
    incident_report_1 = file.read()

# Now, file_content contains the entire content of the file as a string
#print(incident_report_1)

In [78]:
# Specify the file path (make sure it's correct)
file_path = r'C:\Users\dalre\Documents\GitHub\aviation_investigation_model\Transcriptions\transcribe_recording1.txt'

# Open the file in read mode and read its content into a string
with open(file_path, 'r') as file:
    transcription_1 = file.read()

# Now, file_content contains the entire content of the file as a string
#print(transcription_1)

In [79]:
# Specify the file path (make sure it's correct)
file_path = r'C:\Users\dalre\Documents\GitHub\aviation_investigation_model\Conclusions\conclusion_2.txt'

# Open the file in read mode and read its content into a string
with open(file_path, 'r') as file:
    conclusion= file.read()

# Now, file_content contains the entire content of the file as a string
#print(conclusion)

In [80]:
# Specify the file path (make sure it's correct)
file_path = r'C:\Users\dalre\Documents\GitHub\aviation_investigation_model\Data\classification.txt'

# Open the file in read mode and read its content into a string
with open(file_path, 'r') as file:
    classification= file.read()

# Now, file_content contains the entire content of the file as a string
#print(conclusion)

In [82]:
prompt = f"""
Your Task: write a conclusion in 300 words based on the transcription, 
following example output. 
Focus on:
Severity <{classification}> based on the situation.
Runway and operational details.
ATCO actions during the incident.
Consequences and possible investigations.
example output: <{conclusion}>
transcription: <{transcription_1}>
"""

#incident_report:<{incident_report_1}>

In [83]:
from IPython.display import display
display(chat_completion(prompt))

"<**Severity Category B (Significant)**  \n**ATCO Actions & Aircraft Interactions**  \nAT Coordinator's Actions - Managed a near-collision scenario between United Airlines 4253 and American Airlines 3721 involving erroneous taxiing on taxiway Yankee during ramp operations. Some confusion was displayed when Federal Aviation Administration (FAA) inspectors requested a pilot deviation investigation.  \n**Significant Closeness** - The airspace was occupied with multiple conveyances heading towards Runway 1 at varying velocities, increasing operational pressure.  \n**Decision Actions Taken** - The ATCO swiftly issued taxiway detours to both aircraft and ensured relays of updates despite the dynamic situation.  \n**Operational Disruption** - Albeit minimal, the episode occurred amid bustling operations.  \n**Adverse Weather** - The prevailing winds at twelve knots, gusting to seventeen, remained moderate, not posing significant risk during this traffic management effort.>\n\n<**Conclusion** 

Conclusion\nSeverity Category B (Significant) for the incident involving United 326 and Skywest 3721 during a busy operational period at KBKX illustrates substantial operational challenges and human factors contributing to a critical moment where procedural adherence and communication were crucial.\n\nRunway and Operational Details:\nRunway 15 and 8 were in use, as indicated by the ATCO (Air Traffic Control) instructions directing United 326 for takeoff on runway 15 and Skywest 3721 to hold on crossing runway 15.\n\nATCO Actions During the Incident:\nATC communicated openly and promptly with the flight crews involved, showcasing the controller's attention to procedure and the preservation of safety. However, the incident occurred due to a miscommunication regarding the hold instructions to Skywest 3721, leading to an ambiguous situation where Skywest 3721 crossed the intended runway for United 326.\n\nSkywest 3721 was instructed to hold without clear verbal confirmation, resulting in their initial crossing of the runway in error, but with no immediate collision as Sandra AMF1943 adeptly achieved a go-around. \n\nConsequences and Possible Investigations:\nFollowing the event, ground control directed United 326 and related parties to specific frequencies to clarify the situation, demonstrating a systematic approach to resolving the ambiguity. Although procedural non-compliance occurred, there were no reported injuries, damages, or US FARs (Federal Aviation Regulations) infractions were violated, most importantly, the potential for loss of separation, which was fortunately not realized due to successful aircraft recovery actions.\n\nThe accident and significant close calls on the boundary-condition of Pilot deviation necessitated an evaluation of the communication protocol used between ground control and flight crew. It is suggested that confusion and the critical nature of the case merit a detailed investigation not for a potential radi\noughbreaking near\n\nIn conclusion, the incident can be classified as high severity, with emphasis on:\n\nPilot Noncompliance and runway incurs"